onnxruntime/core/graph/function.cc

```diff
@@ -282,11 +289,17 @@ FunctionImpl::FunctionImpl(const onnxruntime::Graph& graph,
   // as we might make some modifications to the FunctionProto along the way
 
   const auto* node_in_parent_graph = parent_graph_->GetNode(node_index);
+  // For schema defined functions get the version from the node in parent graph.
+  // For the functions which do not have schema defined (model local functions) 
+  // get the since version from the version in opset imports using the domain.
+  auto since_version = node_in_parent_graph->SinceVersion() == -1 
+      ? GetVersionForDomain(node_in_parent_graph->Domain(), body_.MainGraph().DomainToVersionMap()) 
```

Comment:
Seems a bit unintuitive. Why is this needed?

Reply:
For schema defined functions we use the since version from schema and for model local functions we get the since version from opset imports.